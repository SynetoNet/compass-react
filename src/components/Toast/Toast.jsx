import React from "react";
import PropTypes from "prop-types";
import BSToast from "react-bootstrap/Toast";

class Toast extends React.Component {
  static Header = BSToast.Header;
  static Body = BSToast.Body;

  render() {
    return <BSToast {...this.props} />;
  }
}

Toast.propTypes = {
  show: PropTypes.bool,
  onClose: PropTypes.func,
  delay: PropTypes.number,
  autohide: PropTypes.bool
};

Toast.defaultProps = { show: true, delay: 3000, autohide: false };

export default Toast;
