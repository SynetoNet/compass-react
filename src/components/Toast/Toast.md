#### **Default**

```jsx
<Toast>
  <Toast.Header>
    <strong className="mr-auto">Bootstrap</strong>
    <small>11 mins ago</small>
  </Toast.Header>
  <Toast.Body>Hello, world! This is a toast message.</Toast.Body>
</Toast>
```

#### **Dismissible**

```jsx
class ToastExample extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      show: true
    };
  }

  render() {
    const { show } = this.state;
    const toggleShow = () => this.setState({ show: !show });
    return (
      <Toast show={show} onClose={toggleShow}>
        <Toast.Header>
          <img src="holder.js/20x20?text=%20" className="rounded mr-2" alt="" />
          <strong className="mr-auto">Bootstrap</strong>
          <small>11 mins ago</small>
        </Toast.Header>
        <Toast.Body>Woohoo, you're reading this text in a Toast!</Toast.Body>
      </Toast>
    );
  }
}

<ToastExample />;
```

#### **Stacking**

```jsx
class ToastExample extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      showFirst: true,
      showSecond: true
    };
  }

  render() {
    const { showFirst, showSecond } = this.state;
    const toggleShowFirst = () => this.setState({ showFirst: !showFirst });
    const toggleShowSecond = () => this.setState({ showSecond: !showSecond });
    return (
      <>
        <Toast show={showFirst} onClose={toggleShowFirst}>
          <Toast.Header>
            <img
              src="holder.js/20x20?text=%20"
              className="rounded mr-2"
              alt=""
            />
            <strong className="mr-auto">Bootstrap</strong>
            <small>just now</small>
          </Toast.Header>
          <Toast.Body>See? Just like this.</Toast.Body>
        </Toast>
        <Toast show={showSecond} onClose={toggleShowSecond}>
          <Toast.Header>
            <img
              src="holder.js/20x20?text=%20"
              className="rounded mr-2"
              alt=""
            />
            <strong className="mr-auto">Bootstrap</strong>
            <small>2 seconds ago</small>
          </Toast.Header>
          <Toast.Body>Heads up, toasts will stack automatically</Toast.Body>
        </Toast>
      </>
    );
  }
}

<ToastExample />;
```

#### **Autohide**

Hides after 3 seconds

```jsx
import { Button } from "../../index.js";

class ToastExample extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      show: true
    };
  }

  render() {
    const { show } = this.state;
    const toggleShow = () => this.setState({ show: !show });

    return (
      <>
        <Toast show={show} onClose={toggleShow} delay={3000} autohide>
          <Toast.Header>
            <img
              src="holder.js/20x20?text=%20"
              className="rounded mr-2"
              alt=""
            />
            <strong className="mr-auto">Bootstrap</strong>
            <small>11 mins ago</small>
          </Toast.Header>
          <Toast.Body>Woohoo, you're reading this text in a Toast!</Toast.Body>
        </Toast>

        <Button onClick={toggleShow}>Show Toast</Button>
      </>
    );
  }
}

<ToastExample />;
```
