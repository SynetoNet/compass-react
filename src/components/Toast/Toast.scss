@import "../../base/colors";
@import "../../base/variables";

.toast:not(:last-child) {
  margin-bottom: 0.75rem;
}

.toast-success {
  border: 1px solid $semantic-confirm-light-60;
  background-color: $semantic-confirm-light-85;
  color: $neutral-90;

  & .toast-header {
    color: $semantic-confirm-dark-30;
  }
}

.toast-warning {
  border: 1px solid $semantic-warn-light-60;
  background-color: $semantic-warn-light-85;
  color: $semantic-warn-dark-30;

  & .toast-header {
    color: $semantic-warn-dark-30;
  }
}

.toast-info {
  border: 1px solid $semantic-info-light-60;
  background-color: $semantic-info-light-85;
  color: $neutral-90;

  & .toast-header {
    color: $semantic-info-dark-30;
  }
}

.toast-danger {
  border: 1px solid $semantic-danger-light-60;
  background-color: $semantic-danger-light-85;
  color: $semantic-danger-dark-30;

  & .toast-header {
    color: $semantic-danger-dark-30;
  }
}
