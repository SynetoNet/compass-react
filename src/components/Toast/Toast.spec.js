import React from "react";
import Toast from "./Toast";
import renderer from "react-test-renderer";

import { configure, mount, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("Default Toast", () => {
  const defaultExample = (
    <Toast>
      <Toast.Header>
        <strong className="mr-auto">Bootstrap</strong>
        <small>11 mins ago</small>
      </Toast.Header>
      <Toast.Body>Hello, world! This is a toast message.</Toast.Body>
    </Toast>
  );

  test("popover snapshop", () => {
    const component = shallow(defaultExample);
    expect(component).toMatchSnapshot();
  });
});
