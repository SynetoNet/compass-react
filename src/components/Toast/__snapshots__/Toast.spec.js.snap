// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Toast popover snapshop 1`] = `
<Bootstrap(Toast)
  autohide={false}
  delay={3000}
  show={true}
>
  <ToastHeader
    closeButton={true}
    closeLabel="Close"
  >
    <strong
      className="mr-auto"
    >
      Bootstrap
    </strong>
    <small>
      11 mins ago
    </small>
  </ToastHeader>
  <ToastBody>
    Hello, world! This is a toast message.
  </ToastBody>
</Bootstrap(Toast)>
`;
